### PR TITLE
issue-652: do not acquire barriers from the past upon `FlushBytes`

### PR DESCRIPTION
**Pull request by debnatkh**
References (and, hopefully, fixes) #652

The main problem is that FlushBytes acquires collect barrier, which is less than the last collect commit id:

---

1. Consider that there were the following sequence of writes:
```
Write(0,       256 KiB, 'a') -> Blob(commitId = 42)
Write(256 KiB, 256 KiB, 'b') -> Blob(commitId = 43)
Write(512 KiB, 1,       'f') -> FreshBytes(commitId = 44)
Write(0,       256 KiB, 'c') -> Blob(commitId = 45)
```
This will lead to the following file layout: `[ccccccc][bbbbbbb][f]`

2. After execution of the CollectGarbage, all three new blobs will get a KeepFlag and the last collect commit id will be equal to 44
```
CommitId:     41      42        43       44
            Blob(a) Blob(b) FreshBytes Blob(c)
                                          |
                                 LastCollectCommitId
```

3. After execution of the Cleanup operation, the first blob will be marked as garbage

4. Let us execute FlushBytes operation, It will acquire collect barrier, equal to the minimal commitId, associated with FreshBlobs:
https://github.com/ydb-platform/nbs/blob/836a5162f99b998582dc4b476a212619954bfa22/cloud/filestore/libs/storage/tablet/tablet_actor_flush_bytes.cpp#L669-L671

After this acquisition there will be one barrier, equal to 43:
```
CommitId:     41      42        43       44
            Blob(a) Blob(b) FreshBytes Blob(c)
                                |         |
                             Barrier  LastCollectCommitId
```

5. When the next CollectGarbage operation is to be executed, it will choose 42 as a collectCommitId:

https://github.com/ydb-platform/nbs/blob/836a5162f99b998582dc4b476a212619954bfa22/cloud/filestore/libs/storage/tablet/model/garbage_queue.cpp#L227-L228

After it the CollectGarbage request with one new grabage will be sent, leading to a decrease in collectCommitIds sequence: 42 after 44

### CODE REVIEW

**Review by qkrorlqr — APPROVED** [review]

**Review by qkrorlqr — CHANGES_REQUESTED** [review]
Хотя нет, unship, теперь у нас FlushBytes начинает отличаться от Compaction. Compaction генерирует CommitId один раз и использует его как для своих dstBlobs, так и для установки барьера. Не хочется, чтобы FlushBytes отличался от Compaction в этом месте, т.к. по своей сути он делает то же самое, что и Compaction.

File: cloud/filestore/libs/storage/tablet/tablet_actor_flush_bytes.cpp [hunk 1]
```diff
@@ -664,11 +664,7 @@ void TIndexTabletActor::CompleteTx_FlushBytes(
         }
     };
 
-    args.CollectCommitId = Max<ui64>();
 
-    for (const auto& bytes: args.Bytes) {
-        args.CollectCommitId = Min(args.CollectCommitId, bytes.MinCommitId);
-    }
```
Comment (debnatkh) [inline cloud/filestore/libs/storage/tablet/tablet_actor_flush_bytes.cpp:671]: К переменной CollectCommitId нет обращений до 800 строки

File: cloud/filestore/libs/storage/tablet/tablet_actor_flush_bytes.cpp [hunk 2]
```diff
@@ -824,6 +820,10 @@ void TIndexTabletActor::CompleteTx_FlushBytes(
     }
 
     AcquireCollectBarrier(args.CollectCommitId);
+    // TODO(#1923): it may be problematic to acquire the barrier only upon
+    // completion of the transaction, because blobs, that have been read at the
+    // prepare stage, may be tempered with by the time of the transaction
```
Comment (qkrorlqr) [inline cloud/filestore/libs/storage/tablet/tablet_actor_flush_bytes.cpp:825]: tampered with

File: cloud/filestore/libs/storage/tablet/tablet_actor_flush_bytes.cpp [hunk 3]
```diff
@@ -793,15 +789,15 @@ void TIndexTabletActor::CompleteTx_FlushBytes(
     auto dstBlobs = builder.Finish();
     TABLET_VERIFY(dstBlobs);
 
-    auto commitId = GenerateCommitId();
-    if (commitId == InvalidCommitId) {
+    args.CollectCommitId = GenerateCommitId();
```
Comment (qkrorlqr) [inline cloud/filestore/libs/storage/tablet/tablet_actor_flush_bytes.cpp:792]: давай назовем просто args.CommitId

**Review by qkrorlqr — APPROVED** [review]

**Review by yegorskii — APPROVED** [review]

**Pull request merged**